src/rust/inetstack/protocols/tcp/tests/simulator.rs

```diff
@@ -880,6 +1021,38 @@ impl Simulation {
         Ok(())
     }
 
+    /// Runs an outgoing UDP packet.
+    fn run_outgoing_udp_packet(&mut self, udp_packet: &UdpPacket) -> Result<()> {
+        let mut n = 0;
+        let frames = loop {
+            let frames = self.engine.pop_all_frames();
+            if frames.is_empty() {
+                if n > 5 {
```

Comment:
Why 5?

Reply:
Thanks for catching. This code was replicated from TCP path, and the rationale for that constant was empirically chosen by @iyzhang (https://github.com/microsoft/demikernel/blame/93ec4a8d34bd2664f52670d4b9d42002dd22da4a/src/rust/inetstack/protocols/tcp/tests/simulator.rs#L998)

I will add a file constant with a comment on this.